### DISCUSSION TITLE
Calculate common centroid of two adjacent Binary Regions

### DISCUSSION THREAD

**Discussion by tomasmanu** [bonsai-rx/bonsai#1027] [Q&A] [answered]
Hi everyone!
	I'm tracking an animal walking on a spherical treadmill. I'm using two coloured papers above the animal, so that I can calculate its heading direction as the vector defined by the subtraction of the centroids of the two aligned papers. 
![image](https://user-images.githubusercontent.com/69548554/188476909-1f4bb894-74c8-427d-ac82-08e9adb3d55d.png)
The problem is that sometimes one of the papers moves below the pin of the harness, and its Binary Region gets divided in two adjacent regions, causing the centroid to flip. I would like to give some stability to the centroid in that situation. 
My idea is to use SortBinaryRegions to obtain the centroids of the divided BinaryRegion so that I can calculate the 'mean centroid' defined by those two regions together.
![image](https://user-images.githubusercontent.com/69548554/188476782-9cfee635-2114-4e1d-a71f-b5c0e7fb93f8.png)
Can it be a good solution to this problem? However, I'm having problems to deal with the Data types in the Python Transform. How can I deal with the two centroids of the adjacent areas to get one mean centroid? I thought of something like this based on another post:
![image](https://user-images.githubusercontent.com/69548554/188477602-828f98c3-2fac-4033-bc25-e072cca6074e.png)
But I'm having problems with data type conversions, from Point.2f to tuple.
I leave attached the workflow and a video to use as example.
Any help will be appreciated, as well as other approaches different to mine.
Cheers,
Tomás
[workflow_and_video.zip](https://github.com/bonsai-rx/bonsai/files/9490914/workflow_and_video.zip)

**Comment by bruno-f-cruz** [comment, marked as answer]
try `@returns(Tuple[float, float])` instead of `@returns(tuple)`

**Reply by tomasmanu** [reply]
Thanks Bruno! Now it works. In order to see the new mean centroid displayed in the crop window, I converted every output back to open.CV.net.Point.2f, using Point2f, and it worked!
